nemo_curator/classifiers/prompt_task_complexity.py

```diff
@@ -0,0 +1,369 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+from dataclasses import dataclass, field
+from typing import Dict, List, Optional
+
+os.environ["RAPIDS_NO_INITIALIZE"] = "1"
+import numpy as np
+import torch
+import torch.nn as nn
+from crossfit import op
+from crossfit.backend.torch.hf.model import HFModel
+from huggingface_hub import PyTorchModelHubMixin
+from transformers import AutoConfig, AutoModel, AutoTokenizer
+
+from nemo_curator.classifiers.base import (
+    DistributedDataClassifier,
+    _get_suggest_memory_for_classifier,
+)
+from nemo_curator.datasets import DocumentDataset
+
+PROMPT_TASK_COMPLEXITY_IDENTIFIER = "nvidia/prompt-task-and-complexity-classifier"
+
+
+@dataclass
+class PromptTaskComplexityConfig:
+    base_model: str = "microsoft/DeBERTa-v3-base"
+    max_len: int = 512
+    model_output_type: Dict = field(default_factory=dict)
+
+
+class MeanPooling(nn.Module):
+    def __init__(self):
+        super(MeanPooling, self).__init__()
```

Comment:
In Python>3.0 it is recommended to use super() function without any arguments.
```suggestion
        super().__init__()
```

Detailed explanation around it: 

https://stackoverflow.com/a/27134600